api/test/planning-units.e2e-spec.ts

```diff
@@ -0,0 +1,220 @@
+  // });
+
+  /**
+   * Set to true if queue data should be removed after each run (see
+   * `afterEach()` below). Keeping in mind that long-running workers may still
+   * be writing to Redis after tests have run, so there may still be new data
+   * showing up in Redis.
+   */
+
+  beforeAll(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [AppModule],
+    }).compile();
+
+    app = moduleFixture.createNestApplication();
+
+    app.useGlobalPipes(
+      new ValidationPipe({
+        transform: true,
+        whitelist: true,
+        forbidNonWhitelisted: true,
+      }),
+    );
+    await app.init();
+
+    const response = await request(app.getHttpServer())
+      .post('/auth/sign-in')
+      .send({
+        username: E2E_CONFIG.users.basic.aa.username,
+        password: E2E_CONFIG.users.basic.aa.password,
+      })
+      .expect(201);
+
+    jwtToken = response.body.accessToken;
+    /**
+     * Create an organization for all of the projects in the test suite.
+     *
+     * No need to assert much here, though we may want to halt the execution
+     * of the test suite if the organization cannot be created for whatever
+     * reason.
+     */
+    // anOrganization = await OrganizationsTestUtils.createOrganization(
+    //   app,
+    //   jwtToken,
+    //   E2E_CONFIG.organizations.valid.minimal(),
+    // ).then(async (response) => {
+    //   return await Deserializer.deserialize(response);
+    // });
+    /**
+     * we pause the queue so workers will not take any job.
+     */
+    await queueService.planningUnitsQueue.pause();
+  });
+  afterEach(async function () {
+    await queueService.planningUnitsQueue.drain(true);
+  });
+
+  afterAll(async () => {
+    /**
+     * On teardown, delete the organization created for the test suite's
+     * projects.
+     * @debt this will be implemented once pr with this functionality is merged
+     */
+    /**
+     * we resume the paused queue after cleanning the jobs.
+     */
+    await queueService.planningUnitsQueue.resume();
+    await Promise.all([app.close(), queueService.onModuleDestroy()]);
+  });
+
+  describe.only('Planning units', () => {
+    let anOrganization: { id: string; type: 'organizations' };
+    let minimalProject: { id: string; type: 'projects' };
+    let customAreaProject: { id: string; type: 'projects' };
+    let adminAreaProject: { id: string; type: 'projects' };
+
+    /**
+     * @description
+     * this should describe how the sinergy between Project creation
+     * and pu creation works
+     */
+
+    it('Creates an organization', async () => {
+      const response = await request(app.getHttpServer())
+        .post('/api/v1/organizations')
+        .set('Authorization', `Bearer ${jwtToken}`)
+        .send(E2E_CONFIG.organizations.valid.minimal())
+        .expect(201);
+
+      anOrganization = response.body.data;
+
+      expect(anOrganization.type).toBe('organizations');
+    });
```

Comment:
this is something I would discourage to take from my (bad) early examples in other tests 😅 

This is all opinion, but basically at the beginning I was creating "parent" entities in self-contained tests, but in practice we shouldn't be testing that creating an organization works as expected in a test suite focused on planning units - in newer tests I have used some test util classes to wrap organization/project/scenario creation and I then create these parent entities as needed in `beforeAll()`, and delete them if needed in `afterAll()`. I will adapt older tests bit by bit, so I am happy for this to stay here if you are ok for me to include this in my cleanup 😅  (same for the deleting projects/org test further down).

Reply:
i know but i was pending that the new ones are in order to incorporate them there. i can leave a todo note and we can refactor this tests later